fv3gfs-util/tests/test_partitioner.py

```diff
@@ -108,150 +107,436 @@ def test_tile_extent_from_rank_metadata(array_extent, array_dims, layout, tile_e
+            False,
+            float(1 / 3),
+            id="96_rank_farbotfarleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            1,
+            (slice(0, 5), slice(0, 2), slice(2, 8)),
+            False,
+            float(1 / 3),
+            id="96_rank_farbotcloseleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            6,
+            (slice(0, 5), slice(2, 8), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_closebotcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            14,
+            (slice(0, 5), slice(14, 16), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_fartopcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            1,
+            (slice(0, 7), slice(3, 7)),
+            True,
+            1.0,
+            id="48_rank_botcloseleft_interface_right_overlap",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            1,
+            (slice(0, 7), slice(2, 7)),
+            True,
+            0.5,
+            id="48_rank_botcloseleft_interface_right_overlap_half_edge",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
+            (13, 13),
+            (2, 4),
+            7,
+            (slice(6, 13), slice(10, 13)),
+            True,
+            0.5,
+            id="48_rank_topfarright_interface_right_overlap_half_edge",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (12, 12),
+            (2, 4),
+            0,
+            (slice(0, 6), slice(0, 2)),
+            True,
+            0.5,
+            id="48_rank_botfarleft_right_overlap_half_edge",
+        ),
     ],
 )
 @pytest.mark.cpu_only
 def test_subtile_slice(
-    array_dims, tile_extent, layout, subtile_index, subtile_slice, overlap
+    array_dims, tile_extent, layout,
+    rank, subtile_slice, overlap, edge_tile_ratio
 ):
-    result = fv3gfs.util.partitioner.subtile_slice(
-        array_dims, tile_extent, layout, subtile_index, overlap
+    partitioner = fv3gfs.util.TilePartitioner(layout, edge_tile_ratio)
+    result = partitioner.subtile_slice(
```

Comment:
nit: in this refactor, could you put `rank` as the third argument so it's in the same position as the two arguments it's replacing (layout and subtile_index)?

Reply:
Done.

Reply:
Rerunning the `pytest` tests, it seems a _lot_ of calls to subtile_slice are used and all of them have the rank as first argument, should I change all the tests to use rank as thirdargument of leave it as is (first argument)?

Here's one example:
`tests/test_tile_scatter_gather.py::test_tile_gather_no_recv_quantity[y,z,x-3-layout0-0-numpy] - TypeError: subtile_slice() got multiple values for argument 'global_dims'`
with the corresponding call:
`subtile_slice = partitioner.subtile_slice(
            rank,
            global_dims=tile_quantity.dims,
            global_extent=tile_quantity.extent,
            overlap=True,
        )`